pkg/network/ebpf/c/runtime/tracer.c

```diff
@@ -401,6 +401,26 @@ int kprobe__tcp_set_state(struct pt_regs* ctx) {
     return 0;
 }
 
+SEC("kprobe/tcp_finish_connect")
+int kprobe__tcp_finish_connect(struct pt_regs *ctx) {
```

Comment:
Good catch. So I guess hook tcp_connect is still needed to map the sock struct to the pid (using the sock struct's pointer as the key).

Reply:
So I added a probe to tcp_connect to map the `sock` to the pid. I also modified the `tcp_close` probe to cleanup map entries in case the TCP connection fails.